compiler/rustc_target/src/abi/call/powerpc64.rs

```diff
@@ -108,7 +67,10 @@ where
     }
 
     let size = arg.layout.size;
-    if size.bits() <= 64 {
+    if is_ret && size.bits() > 128 {
+        // Non-homogeneous aggregates larger than two doublewords are returned indirectly.
+        arg.make_indirect();
```

Comment:
I think this at the very least should use `size.bits() > 64`. Anything between 64bit and 128bit shouldn't be returned using uniform casts. In fact it it may need to be just `if is_ret {}` given that the ELFv2 abi states:

> Aggregates or unions of any length, and character strings of length longer than 8 bytes, will be returned in a storage buffer allocated by the caller.

Reply:
> AFAIK the quote you've quoted is from the ELFv1 ABI (available [here](https://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.pdf)); the ELFv2 ABI (available [here](https://openpowerfoundation.org/specifications/64bitelfabi/)) states:

My bad.

> I'm unsure what you mean regarding not using uniform casts for sizes larger than 64 bits, could you clarify? Clang performs the same casts ([example](https://godbolt.org/z/qhqK4zbT3)).

Never mind. That was based on me accidentally reading the ELFv1 ABI.